odl/contrib/README.md

```diff
@@ -0,0 +1,98 @@
+
+**Important:** Responsibility for code that you contribute will remain with you. This means in particular that issues on the tracker about your module will be assigned to you. Of course, others will try to help if necessary.
+
+**Note:** If you consider your code to be useful for the main library, please consider filing a pull request anyway. We try to be as helpful as possible in our reviews and make suggestions, so that getting the code ready for merging doesn't take too long.
+Note also that contributed modules that attract significant attention and user adoption will be considered for core ODL.
+
+Here are some guidelines for contributors regarding the scope and the rules of this platform.
+
+
+## Scope
+
+**Yes:**
+
+- Generic code that would fit into main ODL but is still experimental,
+- Special-purpose code that is not generic enough to be a fit for main ODL (e.g. [figures of merit](https://github.com/odlgroup/odl/pull/1018) or application-specific code),
+- Extended, more complex examples than [the ones in main ODL](https://github.com/odlgroup/odl/tree/master/examples).
+
+**No:**
+
+- Code for specialized algorithms written specifically for your latest paper,
+- Large files like images, datasets etc.,
+- Huge amounts of code with questionable usefulness,
+- Code under licenses other than MPL-2.0.
+
+For code of the type "your latest paper", we recommend using a separate Github repository like [this one](https://github.com/kohr-h/variable_lp_paper) and making a pull request to add your project to the "Projects using ODL" section of the main README.
+
+
+## Procedure
+
+Since you do not have write access to the target location in the ODL repository, do the initial work on the code on [your personal fork](https://odl.readthedocs.io/dev/gitwash/forking_hell.html#forking). Within this fork, add your contribution to the `odl/contrib` directory. When the code is ready, the steps are as follows:
+
+- Submit a pull request. If you would like an in-depth review, please indicate so in writing or by choosing a reviewer on the Github pull request page.
+- We will check the pull request as a whole and (optionally) the code. Note that your tests must pass and the code must adhere to [PEP8](https://www.python.org/dev/peps/pep-0008/) for Travis CI to go green.
+- To keep the history clean and to avoid accidental additions and deletions that bloat the repository, we will squash your pull request into a single commit that summarizes the whole addition. You can choose to do that yourself and write [a good commit message](https://odl.readthedocs.io/dev/gitwash/development_workflow.html#the-commit-message).
+- If no in-depth review was asked for, we will simply merge the PR if everything is okay. Otherwise, merging is delayed until you indicate that you have modified the code according to review.
+
+That's all.
+
+
+## Package layout
```

Comment:
Yes, currently nothing in `contrib` is imported when you `import odl`. I tested that with putting a sleep in the module code. It only applies when you explicitly `from odl.contrib import mylib`.
So lazy importing happens at the level of `odl/contrib`, and I guess that's good enough. With `odl/contrib` ignored by `pytest` by default this shouldn't pose any problems whatsoever. But I'll test this.